src/matching/quoteMatcher.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * QuoteMatcher matches quoted strings, respecting escaped quotes (\") and frieds
+ */
+export class QuoteMatcher {
+  static escapeChar = "\\";
+
+  static findOpening(char: string, corpus: string, start: number): number {
+    // first, search backwards to see if we could be inside a quote
```

Comment:
It's a bit tricky, and I probably didn't comment enough. From my observed behavior of Vim, this algorithm works as expected. First, you check if the opening quote is behind you (the first iteration of this loop). If you find a valid "opening" quote, stop processing and look for the closing quote after the opening position.

If this loop **fails**, then search forwards to see if the quoted string starts **after** the current position.